### PR TITLE
MAT-88 + MAT-92 start tagging stripe payment intent with useful metadata

### PR DESCRIPTION
**Pull request by tbg-jasonfung**
- MAT-88 start tagging stripe payment intent with useful metadata
- MAT-92 optionally set donation fields if available

### CODE REVIEW

File: src/Application/Actions/Donations/Create.php [hunk 1]
```diff
@@ -91,6 +91,13 @@ protected function action(): Response
                     // See https://stripe.com/docs/api/payment_intents/object
                     'amount' => (100 * $donation->getAmount()),
                     'currency' => 'gbp',
+                    'metadata' => [
+                        'campaignId' => $donation->getCampaign()->getSalesforceId(),
```
Comment (NoelLH) [inline src/Application/Actions/Donations/Create.php:95]: @tbg-jasonfung I think campaign name as mentioned on the description would also be handy.

Thinking further it's probably good to include all 4 (charity + campaign) (names + IDs) and comms flags so we can see these all at a glance.

File: src/Application/Actions/Donations/Create.php [hunk 2]
```diff
@@ -91,6 +91,13 @@ protected function action(): Response
...
+                        'charityName' => $donation->getCampaign()->getCharity()->getName(),
+                        'env' => getenv('APP_ENV'),
```
Comment (NoelLH) [inline src/Application/Actions/Donations/Create.php:97]: I would call this `environment` here for clarity unless we're short on space.

File: src/Application/Actions/Donations/Create.php [hunk 3]
```diff
@@ -91,6 +91,13 @@ protected function action(): Response
...
+                        'isGiftAid' => $donation->isGiftAid(),
```
Comment (NoelLH) [inline src/Application/Actions/Donations/Create.php:98]: It's probably good to have `coreDonationGiftAid` and `tbgTipGiftAid` and set them both to the main flag's value initially, as it sounds like we plan to split the declaration soon.

File: src/Application/Actions/Donations/Create.php (hunk 1 above)
Comment (tbg-jasonfung) [reply src/Application/Actions/Donations/Create.php:95]: @NoelLH I don't think we're sending campaign name through the API, should we start doing this?

Comment (NoelLH) [reply src/Application/Actions/Donations/Create.php:95]: @tbg-jasonfung we are storing the name, have a look at the model. You may need to add a getter if this is the first use. IDEs can generally generate this for you.

File: src/Application/Actions/Donations/Create.php [hunk 4]
```diff
@@ -68,6 +68,27 @@ protected function action(): Response
 
         try {
             $donation = $this->donationRepository->buildFromApiRequest($donationData);
+
+            // Optionally set these donation fields if available
+            if (isset($donationData->billingPostalAddress)) {
```
Comment (NoelLH) [inline src/Application/Actions/Donations/Create.php:73]: We'll need to revisit whether we actually need to collect billing and postal address separately for Gift Aid + card processing purposes. This doesn't align with the WIP for the frontend but may still be the best option.

File: src/Application/Actions/Donations/Create.php [hunk 5]
```diff
@@ -68,6 +68,27 @@ protected function action(): Response
 
         try {
             $donation = $this->donationRepository->buildFromApiRequest($donationData);
+
+            // Optionally set these donation fields if available
```
Comment (NoelLH) [inline src/Application/Actions/Donations/Create.php:72]: These setters are all doing what `buildFromApiRequest()` describes as its action, so should be in that method.

We're also missing country code.

File: src/Application/Actions/Donations/Create.php [hunk 6]
```diff
@@ -68,6 +68,27 @@ protected function action(): Response
...
+            if (isset($donationData->billingPostalAddress)) {
+                $donation->setDonorPostalAddress((string) $donationData->billingPostalAddress);
```
Comment (NoelLH) [inline src/Application/Actions/Donations/Create.php:74]: The values which are strings should not need casting to string. What is the use case here? We might also want to consider whether blank string is helpful and test strings for `empty()` instead if it's better to leave the fields null.

File: src/Domain/DonationRepository.php [hunk 7]
```diff
@@ -107,6 +107,31 @@ public function buildFromApiRequest(DonationCreate $donationData): Donation
         $donation->setCharityComms($donationData->optInCharityEmail);
         $donation->setTbgComms($donationData->optInTbgEmail);
 
+        // Optionally set these donation fields if available
+        if (!empty($donationData->billingPostalAddress)) {
```
Comment (tbg-jasonfung) [inline src/Domain/DonationRepository.php:111]: @NoelLH I've just done an optional check for billingPostalAddress for now, we can revisit once we start sending the separate GA claim address
